lib/modules/manager/pub/artifacts.ts

```diff
@@ -63,9 +52,11 @@ export async function updateArtifacts({
       );
     }
 
+    const oldLazyPubspecLock = lazyParsePubspeckLock(oldLockFileContent);
+    const oldPubspecLock = oldLazyPubspecLock.getValue();
```

Comment:
inline this, so the lockfile is only read when no constraints are manually configured. that's the idea behind the lazy.